pkg/build/registry/buildconfig/strategy.go

```diff
@@ -21,10 +21,14 @@ import (
 type strategy struct {
 	runtime.ObjectTyper
 	names.NameGenerator
+	legacy bool
 }
 
 // Strategy is the default logic that applies when creating and updating BuildConfig objects.
-var Strategy = strategy{kapi.Scheme, names.SimpleNameGenerator}
+var Strategy = strategy{kapi.Scheme, names.SimpleNameGenerator, false}
+
+// LegacyStrategy is the default logic that applies when creating and updating BuildConfig objects.
```

Comment:
i'll expand the comment.  applies is correct/intentional.  "this is the rule that applies in this situation"